test/checkkeys.c

```diff
@@ -68,13 +68,19 @@ static void PrintKey(SDL_keysym *sym, int pressed)
 			printf(" (^%c)", sym->unicode+'@');
 		} else {
 #ifdef UNICODE
-			printf(" (%c)", sym->unicode);
-#else
+			printf(" '%c' (0x%.4X)", sym->unicode, (int)sym->unicode);
+#elif defined(_WIN32)
 			/* This is a Latin-1 program, so only show 8-bits */
 			if ( !(sym->unicode & 0xFF00) )
-				printf(" (%c)", sym->unicode);
+				printf(" '%c' (0x%.4X)", sym->unicode, (int)sym->unicode);
 			else
-				printf(" (0x%X)", sym->unicode);
+				printf(" (0x%.4X)", (int)sym->unicode);
+#else /* other platforms than Windows hopefully use UTF-8 for 8bit chars */
+			Uint32 utf32str[2] = { sym->unicode, 0 };
```

Comment:
@ccawley2011: this broke os/2 build using watcom:
`checkkeys.c(79): Error! E1054: Expression must be constant`

I can do the following to build the object:
```diff
diff --git a/test/checkkeys.c b/test/checkkeys.c
index 2df1d90..86ead56 100644
--- a/test/checkkeys.c
+++ b/test/checkkeys.c
@@ -76,9 +76,12 @@ static void PrintKey(SDL_keysym *sym, int pressed)
 			else
 				printf(" (0x%.4X)", (int)sym->unicode);
 #else /* other platforms than Windows hopefully use UTF-8 for 8bit chars */
-			Uint32 utf32str[2] = { sym->unicode, 0 };
 			const char* utf32type = (SDL_BYTEORDER == SDL_LIL_ENDIAN) ? "UTF-32LE" : "UTF-32BE";
-			char* utf8str = SDL_iconv_string("UTF-8", utf32type, (const char*)utf32str, 2*4);
+			char* utf8str;
+			Uint32 utf32str[2];
+			utf32str[0] = sym->unicode;
+			utf32str[1] = 0;
+			utf8str = SDL_iconv_string("UTF-8", utf32type, (const char*)utf32str, 2*4);
 			printf(" '%s' (0x%.4X)", utf8str, (int)sym->unicode);
 			SDL_free(utf8str);
 #endif
```

... but who knows for what reason, the os/2 build of SDL12.dll doesn't include SDL_iconv.c
I can add it, of course, but I'd like to be sure about the ifdef correctness here.

Reply:
I guess it'd be best to leave os/2 alone as it was? Like:
```diff
diff --git a/test/checkkeys.c b/test/checkkeys.c
index 2df1d90..ec5a4cd 100644
--- a/test/checkkeys.c
+++ b/test/checkkeys.c
@@ -70,5 +70,5 @@ static void PrintKey(SDL_keysym *sym, int pressed)
 #ifdef UNICODE
 			printf(" '%c' (0x%.4X)", sym->unicode, (int)sym->unicode);
-#elif defined(_WIN32)
+#elif defined(_WIN32) || defined(__OS2__)
 			/* This is a Latin-1 program, so only show 8-bits */
 			if ( !(sym->unicode & 0xFF00) )
```

Reply:
I went ahead and applied these two patches.